src/date/parse.js

```diff
@@ -29,10 +28,33 @@ return function( value, tokens, properties ) {
 		MINUTE = 4,
 		SECOND = 5,
 		MILLISECONDS = 6,
-		date = new Date(),
+    calendar = Globalize.calendars[properties.calendar],
 		truncateAt = [],
 		units = [ "year", "month", "day", "hour", "minute", "second", "milliseconds" ];
 
+	// Assumptions:
+	// If only times are set, use today's date.
+	// If era is not set, use today's era
+	// If year is not set, use today's year
+	// If neither month nor date are set, use the beginning of the year
+	//   Using the beginning of the year is an exception to all these other
+	//   rules. Why not today's date? The technical report
+	//   http://www.unicode.org/reports/tr35/tr35-dates.html doesn't specify
+	// If month is not set but date is, use today's month
+
+	date = new Date();
+	gdate = new calendar( date );
+	ret = {
+		era: gdate.getEra(),
+		year: gdate.getYear(),
+		month: undefined,
+		date: undefined,
+		hour: date.getHours(),
+		minute: date.getMinutes(),
+		second: date.getSeconds(),
+		milliseconds: 0
+	};
```

Comment:
Can we keep each field in its own local variable (not a ret property), like the example below. Because, we're not taking any benefit from having it inside `ret`. In fact, `ret.` becomes repetitive below.

```js
era = gdate.getEra();
year = ...
...
```